iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/relational/TableScanOperator.java

```diff
@@ -307,17 +309,27 @@ private void prepareForNextDevice() {
     }
   }
 
-  private AlignedSeriesScanUtil constructAlignedSeriesScanUtil(DeviceEntry deviceEntry) {
+  private void constructAlignedSeriesScanUtil() {
+    DeviceEntry deviceEntry;
+
+    if (this.deviceEntries.isEmpty() || this.deviceEntries.get(this.currentDeviceIndex) == null) {
```

Comment:
throw Exception for this.deviceEntries.get(this.currentDeviceIndex) == null

Reply:
No need to construct `AlignedFullPath` and `seriesScanUtil` for `this.deviceEntries.isEmpty()`

Reply:
fix